### ISSUE TITLE
Oh noes miss firing in an object method.

### ISSUE THREAD

**Issue by brew4321** [Khan/live-editor#535]
Take a look at this simplified example program: https://www.khanacademy.org/computer-programming/oh-noes-misfire/6562709088370688
I usually run into this problem when I carelessly cut and paste code around, and forget to bring the variables that the code snippet depends on. The problem is that oh noes instead of reminding me that (for example) "bw is not defined" gets confused and thinks pos should be pop..
"Oh noes!
Did you mean to type pop instead of pos?
Show me where"

**Comment by kevinbarabash** [comment]
Nice find.  The problem is how we check spelling right now.  Currently we run a regex to split a line up into words and then check each of those words against global methods.  We only check spelling on a line when jshint has reported an error on that line.  The jshint error contains the name of the undefined variable so we should probably be passing that the spell checker and only checking against that identifier instead of all identifiers in the line.

**Comment by brew4321** [comment]
I thought it might be something like that, but I didn't know enough about it to really know.. hey at least the way it's broken it should always point to the right line! :)

**Comment by DozenalFTW** [comment]
What's really bad is that it doesn't even matter if it's a variable undefined at all:
```javascript
if(this.pos.x+bw 10)]{
```
still throws the same error.

**Comment by kevinbarabash** [comment]
Yup... if there's an error on a line it blindly checks for spelling mistakes.  We can definitely make this better.

**Comment by DozenalFTW** [comment]
**Bump**
Any luck? Can you report the JSON of the JSHINT error message so that I can see what the undefined variable is stored in?

**Comment by kevinbarabash** [comment]
@DozenalFTW I'll try to get to it on the weekend.
